packages/runtime/src/useCases/transport/tokens/UpdateTokenContent.ts

```diff
@@ -0,0 +1,52 @@
+import { Serializable } from "@js-soft/ts-serval";
+import { Result } from "@js-soft/ts-utils";
+import { TokenDTO } from "@nmshd/runtime-types";
+import { TokenController, TokenReference } from "@nmshd/transport";
+import { Inject } from "@nmshd/typescript-ioc";
+import { RuntimeErrors, SchemaRepository, SchemaValidator, TokenReferenceString, UseCase } from "../../common";
+import { TokenMapper } from "./TokenMapper";
+
+export interface UpdateTokenContentRequest {
+    reference: TokenReferenceString;
+    content: any;
+}
+
+class Validator extends SchemaValidator<UpdateTokenContentRequest> {
+    public constructor(@Inject schemaRepository: SchemaRepository) {
+        super(schemaRepository.getSchema("UpdateTokenContentRequest"));
+    }
+}
+
+export class UpdateTokenContentUseCase extends UseCase<UpdateTokenContentRequest, TokenDTO> {
+    public constructor(
+        @Inject private readonly tokenController: TokenController,
+        @Inject validator: Validator
+    ) {
+        super(validator);
+    }
+
+    protected async executeInternal(request: UpdateTokenContentRequest): Promise<Result<TokenDTO>> {
+        let tokenContent;
+        try {
+            tokenContent = Serializable.fromUnknown(request.content);
+        } catch {
+            throw RuntimeErrors.general.invalidTokenContent();
+        }
+
+        const reference = TokenReference.fromTruncated(request.reference);
+
+        const passwordProtection = reference.passwordProtection;
+        if (passwordProtection && !passwordProtection.password) {
+            throw new Error("RuntimeErrors.general.noPasswordProvided()");
```

Comment:
This error code (if it should be one) looks kind of weird:
- Did you add these parentheses on purpose?
- The "RuntimeErrors" prefix is something we don't usually do.

Did you mean to do the following?


```suggestion
            throw new Error(RuntimeErrors.general.noPasswordProvided());
```